crates/oxc_linter/src/rules/eslint/default_case_last.rs

```diff
@@ -0,0 +1,119 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(default-case-last): Enforce default clauses in switch statements to be last")]
+#[diagnostic(severity(warning))]
+struct DefaultCaseLastDiagnostic(#[label("Default clause should be the last clause.")] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct DefaultCaseLast;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforce default clauses in switch statements to be last
+    ///
+    /// ### Why is this bad?
+    /// A switch statement can optionally have a default clause.
+    /// If present, it’s usually the last clause, but it doesn’t need to be. It is also allowed to put the default clause before all case clauses, or anywhere between. The behavior is mostly the same as if it was the last clause. The default block will be still executed only if there is no match in the case clauses (including those defined after the default), but there is also the ability to “fall through” from the default clause to the following clause in the list. However, such flow is not common and it would be confusing to the readers.
+    /// Even if there is no “fall through” logic, it’s still unexpected to see the default clause before or between the case clauses. By convention, it is expected to be the last clause.
+    /// If a switch statement should have a default clause, it’s considered a best practice to define it as the last clause.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// switch (foo) {
+    ///     default:
+    ///         bar();
+    ///         break;
+    ///     case "a":
+    ///         baz();
+    ///         break;
+    /// }
+    ///
+    /// switch (foo) {
+    ///     case 1:
+    ///         bar();
+    ///         break;
+    ///     default:
+    ///         baz();
+    ///         break;
+    ///     case 2:
+    ///         qux();
+    ///         break;
+    /// }
+    /// ```
+    DefaultCaseLast,
+    correctness
```

Comment:
I wonder whether this should be a `style` rule?

Reply:
Sorry, I didn't notice it. I will check it later.